realm/realm-library/src/main/java/io/realm/RealmAnyOperator.java

```diff
@@ -406,7 +406,7 @@ protected NativeRealmAny createNativeRealmAny() {
 
     @Override
     Class<?> getTypedClass() {
-        return clazz;
+        return RealmObjectProxy.class.isAssignableFrom(clazz) ? clazz.getSuperclass() : clazz;
```

Comment:
Why this change?

Reply:
It is a bug fix. If you initialize a RealmAny with a Managed object it would return the Proxy class as the value class instead of the base model class. It is an issue when you try to compare the value class against a model class like it is done in the module mediator.

I am adding a test case to validate the behavior.